integration-tests/src/test/kotlin/org/stellar/anchor/platform/AnchorPlatformIntegrationTest.kt

```diff
@@ -2,149 +2,227 @@ package org.stellar.anchor.platform
+    envMap["assets.value"] = getResourceFilePath(envMap["assets.value"]!!)
+    envMap["sep1.toml.value"] = getResourceFilePath(envMap["sep1.toml.value"]!!)
+
+    // Start servers
+    val jobs = mutableListOf<Job>()
+    val scope = CoroutineScope(Dispatchers.Default)
+    jobs += scope.launch { ServiceRunner.startKotlinReferenceServer(false) }
+    jobs += scope.launch { runningServers.add(ServiceRunner.startAnchorReferenceServer()) }
+    jobs += scope.launch { runningServers.add(ServiceRunner.startStellarObserver(envMap)) }
+    jobs += scope.launch { runningServers.add(ServiceRunner.startSepServer(envMap)) }
+    jobs.forEach { it.join() }
+
+    // Query SEP-1
+    val toml =
+      Sep1Helper.parse(
+        resourceAsString("http://localhost:${config.sepServerPort}/.well-known/stellar.toml")
+      )
+
+    // Create Sep10Tests
+    sep10Tests = Sep10Tests(toml)
+
+    // Get JWT
+    val jwt = sep10Tests.sep10Client.auth()
+
+    sep12Tests = Sep12Tests(config, toml, jwt)
+    sep24Tests = Sep24Tests(config, toml, jwt)
+    sep31Tests = Sep31Tests(config, toml, jwt)
+    sep38Tests = Sep38Tests(config, toml, jwt)
+    platformApiTests = PlatformApiTests(config, toml, jwt)
+    callbackApiTests = CallbackApiTests(config, toml, jwt)
+    stellarObserverTests = StellarObserverTests()
+  }
 
-      toml =
-          Sep1Helper.parse(
-              resourceAsString("http://localhost:$SEP_SERVER_PORT/.well-known/stellar.toml"))
+  fun stopServers() {
+    runningServers.forEach { SpringApplication.exit(it) }
+    org.stellar.reference.stop()
+  }
 
-      Sep10Tests.setup()
+  fun stopDocker() {
+    docker.afterAll(null)
+  }
+}
 
-      if (!::jwt.isInitialized) {
-        jwt = sep10Client.auth()
-      }
+@TestMethodOrder(MethodOrderer.OrderAnnotation::class)
+class AnchorPlatformIntegrationTest :
+  DefaultIntegrationTest(
+    TestConfig(
+      name = "default",
+      referenceServerPort = 8081,
+      sepServerPort = 8080,
+      observerHealthServerPort = 8083,
+      sep10JwtSecret = "secret",
+      sep24InteractiveUrlJwtSecret = "secret_sep24_interactive_url_jwt_secret",
+      sep24MoreInfoUrlJwtSecret = "secret_sep24_more_info_url_jwt_secret",
+      platformToAnchorSecret = "myPlatformToAnchorSecret"
+    )
+  ) {
+  companion object {
+    private val instance = AnchorPlatformIntegrationTest()
 
-      Sep12Tests.setup()
-      Sep24Tests.setup()
-      Sep31Tests.setup()
-      Sep38Tests.setup()
-      PlatformApiTests.setup()
+    @BeforeAll
+    @JvmStatic
+    fun setup() {
+      instance._setUp()
     }
 
     @AfterAll
     @JvmStatic
-    fun afterAllTests() {
-      if (shouldStartServers) stopServers()
-      if (shouldStartDockerCompose) stopDocker()
-    }
-
-    fun stopServers() {
-      runningServers.forEach { SpringApplication.exit(it) }
-      org.stellar.reference.stop()
-    }
-
-    fun stopDocker() {
-      docker.afterAll(null)
+    fun tearDown() {
+      instance._tearDown()
     }
   }
 
   @Test
```

Comment:
Is there a reason tests are split like that?
Why `AnchorPlatformIntegrationTest` is not simply a suite containing all tests?
And all children are simple tests classes (`Sep10Tests`, etc)
Right now it's hard to run 1 test scenario on it's own

Reply:
💡  This PR is still WIP.  I am still re-writing the test frameworks.

Since it is still making major changes, it would save you some efforts if you may review after the PR is ready.

Reply:
I noticed it a while ago, and wanted to check if it's planned to keep the current structure or change it

Reply:
The current structure will be changed.